crates/noirc_evaluator/src/brillig/brillig_ir.rs

```diff
@@ -301,6 +301,22 @@ impl BrilligContext {
             rhs: scratch_register_j,
         });
     }
+
+    /// Emits a modulo instruction against 2**target_bit_size
+    pub(crate) fn cast_instruction(
+        &mut self,
+        destination: RegisterIndex,
+        source: RegisterIndex,
+        target_bit_size: u32,
+    ) {
+        assert!(
+            target_bit_size < 127,
+            "tried to cast to a bit size greater than 126 {target_bit_size}"
+        );
+
+        let modulus = self.make_constant(Value::from(1_u128 << target_bit_size));
+        self.modulo_instruction(destination, source, modulus, target_bit_size + 1, false);
```

Comment:
Why is this target_bit_size + 1 ?

Reply:
Since the modulus is 1 << bit_size, the minimum number of bits that can contain the modulus without reducing it to zero is bit_size + 1. We could use the max bit_size also (127)